File: DSA/Ayush_Mishra_590014844/Day_18_Question_3.c
#include <stdio.h>

void sort_colors(int arr[], int n) {
    int low = 0, mid = 0, high = n - 1, temp;
    while(mid <= high) {
        switch(arr[mid]) {
            case 0:
                temp = arr[low];
                arr[low] = arr[mid];
                arr[mid] = temp;
                low++; mid++;
                break;
            case 1:
                mid++;
                break;
            case 2:
                temp = arr[mid];
                arr[mid] = arr[high];
                arr[high] = temp;
                high--;
                break;
        }
    }
}

int main() {
    int arr[100], n, i;
    printf("Enter number of elements for color sort (only 0, 1, 2): ");
    scanf("%d", &n);
    printf("Enter elements: ");
    for(i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    sort_colors(arr, n);

    printf("Sorted array: ");
    for(i = 0; i < n; i++)
        printf("%d ", arr[i]);
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_18_Question_3.py
def sort_colors(arr):
    low, mid, high = 0, 0, len(arr) - 1
    while mid <= high:
        if arr[mid] == 0:
            arr[low], arr[mid] = arr[mid], arr[low]
            low += 1
            mid += 1
        elif arr[mid] == 1:
            mid += 1
        else:
            arr[mid], arr[high] = arr[high], arr[mid]
            high -= 1
    return arr

arr = list(map(int, input("Enter 0s, 1s, and 2s for color sort (space-separated): ").split()))
print("Sorted array:", sort_colors(arr))
